rasa/server.py

```diff
@@ -227,27 +227,85 @@ def event_verbosity_parameter(
         )
 
 
-async def get_tracker(
+async def get_tracker_with_session_start(
     processor: "MessageProcessor", conversation_id: Text
 ) -> DialogueStateTracker:
-    """Get tracker object from `MessageProcessor`."""
-    tracker = await processor.get_tracker_with_session_start(conversation_id)
-    _validate_tracker(tracker, conversation_id)
+    """Get tracker object from `MessageProcessor` and update the conversation session.
 
-    # `_validate_tracker` ensures we can't return `None` so `Optional` is not needed
-    return tracker  # pytype: disable=bad-return-type
+    Args:
+        processor: An instance of `MessageProcessor`.
+        conversation_id: Conversation ID to fetch the tracker for.
 
+    Returns:
+        The tracker for `conversation_id` with an updated conversation session.
+    """
+    return await processor.get_tracker_with_session_start(conversation_id)
```

Comment:
I don't think it's intuitive that a `get_` function has side effects not only on the objects but even on the persistence layer. However, somewhat out of the scope of this PR 😬